### PR TITLE
feat: adds a refresh button

### PR DESCRIPTION
**Pull request by korve**
### Description:

this commit introduces a refresh button to every page in the CoreHome module. When the user clicks the button, the data of the current page is refreshed without reloading the page.

I've added a shortcut. So if this PR makes it into production, i think the documentation needs to be upgraded at  https://matomo.org/faq/how-to/faq_24340/

Refs #11047

https://user-images.githubusercontent.com/4459994/170113211-bf66805a-0fb5-4c28-89c5-f9e81d52d7ed.mp4

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/CoreHome/templates/_headerMessage.twig [hunk 1]
```diff
@@ -58,8 +58,10 @@
     </div>
 </div>
 
+<span class="icon icon-refresh" title="{{ 'CoreHome_Refresh'|translate }}"></span>
```
Comment (peterhashair) [inline plugins/CoreHome/templates/_headerMessage.twig]:
```suggestion
<span class="icon icon-reload" style="margin-right: 10px;" title="{{ 'CoreHome_Refresh'|translate }}"></span>
```

File: plugins/CoreHome/templates/_headerMessage.twig [hunk 2]
```diff
@@ -58,8 +58,10 @@
...
 <span class="icon icon-arrowup"></span>
 <div style="clear:right"></div>
 {% else %}
+<span class="icon icon-refresh" title="{{ 'CoreHome_Refresh'|translate }}"></span>
```
Comment (peterhashair) [inline plugins/CoreHome/templates/_headerMessage.twig]:
```suggestion
<span class="icon icon-reload" style="margin-right: 10px;" title="{{ 'CoreHome_Refresh'|translate }}"></span>
```

**Review by peterhashair — COMMENTED** [review]
should we use reload icon, I believe other places are using that one, a couple of test screenshots needs to be updated which I can do, plugins/CoreHome/lang/en.json should be reverted, it does translate automatically by the system but other than that looks good.

**Comment by peterhashair** [comment]
@justinvelluppillai I feels like this one is good to merge, I will fix the UI tests and lang in another PR.

**Comment by justinvelluppillai** [comment]
Sounds good @peterhashair - you can go ahead and do that. It might be better to create a branch off this branch and fix the UI tests and remove the de translation before merging if you know how.

**Comment by peterhashair** [comment]
@justinvelluppillai sounds good.

File: plugins/CoreHome/javascripts/refresh-button.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    Matomo.postEvent('loadPage', hashParsed.category, hashParsed.subcategory);
+  }
+
+  function isCoreHomeModuleActive() {
+    var search = window.CoreHome.MatomoUrl.parse(window.location.search.slice(1));
+    return search.module === 'CoreHome';
+  }
+
+  if (isCoreHomeModuleActive()) {
+    iconRefresh.show();
```
Comment (sgiehl) [inline plugins/CoreHome/javascripts/refresh-button.js:23]: Just a small note on that. Using `display:none` as default, but calling `show()` here, causes the icon to be visible also in mobile view. Which shouldn't be intended. The reason is, that `show()` will add a `display:inline`, which overwrites the `display:none` that is set for smaller resolutions. That looks a bit ugly.
A better possibility is to always add the class `hidden` and remove it if needed.  See https://github.com/matomo-org/matomo/blob/4.x-dev/tests/UI/expected-screenshots/UIIntegrationTest_dashboard5_mobile.png
@peterhashair you should have actually seen that before updating the screenshots.
I will create a PR to fix that.

Comment (sgiehl) [reply plugins/CoreHome/javascripts/refresh-button.js:23]: Just saw there was already a fix for it in https://github.com/matomo-org/matomo/pull/19289, so nvm.

**Pull request merged**